merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/PostgresResultsCellRepository.java

```diff
@@ -117,17 +117,22 @@ public Optional<ResultsProtocol.OwnerRole> claim(final PlanId planId, final Long
   }
 
   @Override
-  public Optional<ResultsProtocol.ReaderRole> lookup(final PlanId planId) {
+  public Optional<ResultsProtocol.ReaderRole> lookup(final PlanId planId, final Optional<SimulationDatasetId> simulationDatasetId) {
     try (final var connection = this.dataSource.getConnection()) {
       final var simulation = getSimulation(connection, planId);
-      final var datasetRecord = lookupSimulationDatasetRecord(
+      final Optional<SimulationDatasetRecord> datasetRecord;
+      if(simulationDatasetId.isPresent()) {
```

Comment:
Formatting nit: missing space
```suggestion
      if (simulationDatasetId.isPresent()) {
```

Reply:
Done